integration-tests/tests/pg.test.ts

```diff
@@ -2384,6 +2387,500 @@ test.serial('select from enum', async (t) => {
+			)
+	`);
+
+	const insertedDate = new Date('2022-01-01 20:00:00.123-04'); // used different time zones, internally is still UTC
+	const insertedDate2 = new Date('2022-01-02 04:00:00.123+04'); // They are both the same date in different time zones
+
+	// 1. Insert date as new dates with different time zones
+	await db.insert(table).values([
+		{ timestamp: insertedDate },
+		{ timestamp: insertedDate2 },
+	]);
+
+	// 2, Select and compare both dates
+	const result = await db.select().from(table);
+
+	t.deepEqual(result[0]?.timestamp.getTime(), result[1]?.timestamp.getTime());
+
+	await db.execute(sql`drop table if exists ${table}`);
+});
+
+test.serial('all date and time columns without timezone first case mode string', async (t) => {
+	const { db } = t.context;
+
+	const table = pgTable('all_columns', {
+		id: serial('id').primaryKey(),
+		timestamp: timestamp('timestamp_string', { mode: 'string', precision: 6 }).notNull(),
+	});
+
+	await db.execute(sql`drop table if exists ${table}`);
+
+	await db.execute(sql`
+		create table ${table} (
+					id serial primary key,
+					timestamp_string timestamp(6) not null
+			)
+	`);
+
+	// 1. Insert date in string format without timezone in it
+	await db.insert(table).values([
+		{ timestamp: '2022-01-01 02:00:00.123456' },
+	]);
+
+	// 2, Select in string format and check that values are the same
+	const result = await db.select().from(table);
+
+	t.deepEqual(result, [{ id: 1, timestamp: '2022-01-01 02:00:00.123456' }]);
+
+	// 3. Select as raw query and check that values are the same
+	const result2 = await db.execute<{
+		id: number;
+		timestamp_string: string;
+	}>(sql`select * from ${table}`);
+
+	t.deepEqual(result2.rows, [{ id: 1, timestamp_string: '2022-01-01 02:00:00.123456' }]);
+
+	await db.execute(sql`drop table if exists ${table}`);
+});
+
+test.serial('all date and time columns without timezone second case mode string', async (t) => {
+	const { db } = t.context;
+
+	const table = pgTable('all_columns', {
+		id: serial('id').primaryKey(),
+		timestamp: timestamp('timestamp_string', { mode: 'string', precision: 6 }).notNull(),
+	});
+
+	await db.execute(sql`drop table if exists ${table}`);
+
+	await db.execute(sql`
+		create table ${table} (
+					id serial primary key,
+					timestamp_string timestamp(6) with time zone not null
+			)
+	`);
+
+	// 1. Insert date in string format with timezone in it
+	await db.insert(table).values([
+		{ timestamp: '2022-01-01T02:00:00.123456-02' },
+	]);
+
+	// 2, Select as raw query and check that values are the same
+	const result = await db.execute<{
+		id: number;
+		timestamp_string: string;
+	}>(sql`select * from ${table}`);
+
+	// Please notice that postgres will transform the date to UTC and it saves it in UTC.
+	// even if set without time zone, just becuase it was provided a timezone offset, it will transform it to UTC
+	t.deepEqual(result.rows, [{ id: 1, timestamp_string: '2022-01-01 04:00:00.123456+00' }]);
```

Comment:
As Postgres [docs](https://www.postgresql.org/docs/current/datatype-datetime.html) states

> In a literal that has been determined to be timestamp without time zone, PostgreSQL will silently ignore any time zone indication. That is, the resulting value is derived from the date/time fields in the input value, and is not adjusted for time zone.

This means that when you insert a string with a timezone into a `timestamp without time zone` type, the timezone should be simply removed without any time conversions. For example, for `2022-01-01T02:00:00.123456-02`, you should receive `2022-01-01 02:00:00.123456` in return

The reason why it happened - you have created `timestamp(6) with time zone` instead of `timestamp(6)`